contracts/acl/ACL.sol

```diff
@@ -417,18 +418,21 @@ contract ACL is IACL, TimeHelpers, AragonApp, ACLHelpers {
     }
 
     function checkOracle(IACLOracle _oracleAddr, address _who, address _where, bytes32 _what, uint256[] _how) internal view returns (bool) {
-        bytes4 sig = _oracleAddr.canPerform.selector;
-
         // a raw call is required so we can return false if the call reverts, rather than reverting
-        bytes memory checkCalldata = abi.encodeWithSelector(sig, _who, _where, _what, _how);
-        uint256 oracleCheckGas = ORACLE_CHECK_GAS;
+        bytes memory checkCalldata = abi.encodeWithSelector(_oracleAddr.canPerform.selector, _who, _where, _what, _how);
 
         bool ok;
-        assembly {
-            ok := staticcall(oracleCheckGas, _oracleAddr, add(checkCalldata, 0x20), mload(checkCalldata), 0, 0)
+        uint256 gasLeftBeforeCall = gasleft();
+
+        if (gasLeftBeforeCall > ERROR_GAS_ALLOWANCE) {
+            uint256 oracleCanPerformGas = gasLeftBeforeCall - ERROR_GAS_ALLOWANCE;
+            assembly {
+                ok := staticcall(oracleCanPerformGas, _oracleAddr, add(checkCalldata, 0x20), mload(checkCalldata), 0, 0)
+            }
         }
 
         if (!ok) {
+            require(gasleft() > max(gasLeftBeforeCall / 64, ERROR_GAS_ALLOWANCE), ERROR_ORACLE_OOG);
             return false;
```

Comment:
I would refactor all this as:
```
        uint256 gasLeftBeforeCall = gasleft();

        // I don't think we even need this, but if we do I would convert it to a require:
        // require(gasLeftBeforeCall > ERROR_GAS_ALLOWANCE, ERROR_ORACLE_OOG);
        assembly {
            ok := staticcall(gasLeftBeforeCall, _oracleAddr, add(checkCalldata, 0x20), mload(checkCalldata), 0, 0)
        }

        if (!ok) {
            uint256 gasLeftAfterCall = gasLeft();
            require(gasLeftAfterCall > gasLeftBeforeCall / 64 && gasLeftAfterCall > ERROR_GAS_ALLOWANCE), ERROR_ORACLE_OOG);
            return false;
        }
```

(and we can get rid of the `max` function below)

Reply:
If we use all the available gas in the `staticcall` is there a chance there won't be enough left over for returning the error (even with the 1/64th rule)? After testing I can't find a `gas` limit where that's the case (eg just a `revert` without an error) but it wasn't exhaustive. Also, due to a `stack too deep` error I can't add any more variables eg `gasLeftAfterCall` unless I remove something eg `oracleCanPerformGas`. So I can do this if you think the second `require` statement can always be reached.

Reply:
> If we use all the available gas in the `staticcall` is there a chance there won't be enough left over for returning the error (even with the 1/64th rule)?

In that case we could leave the commented `require(gasLeftBeforeCall > ...` check, with the proper amount to make sure that 1/64 of that amount is enough.
Anyway, even if that happen, what would be the problem? It would just revert, right? So compared to what would happen later in `require(gasLeftAfterCall >...` we would just miss the error message? Or am I missing something?

Reply:
Yes we would just miss the error message. @sohkai do you have a preference? 

I would like to remove the `max()` function but I can't unless I remove the `oracleCanPerformGas` variable (or some unrelated variable somewhere) which would mean passing all of the gas to the function.